src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/reporting/Branch.sol';
+import 'ROOT/reporting/ReportingToken.sol';
+import 'ROOT/reporting/ReputationToken.sol';
+import 'ROOT/reporting/DisputeBondToken.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/Cash.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/factories/MapFactory.sol';
+import 'ROOT/factories/ShareTokenFactory.sol';
+import 'ROOT/factories/ReportingTokenFactory.sol';
+import 'ROOT/factories/DisputeBondTokenFactory.sol';
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/libraries/token/ERC20Basic.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/libraries/math/SafeMathInt256.sol';
+
+
+contract Market is DelegationTarget, Typed, Initializable {
+    using SafeMathUint256 for uint256;
+    using SafeMathInt256 for int256;
+
+    enum ReportingState {
+        AUTOMATED,
+        LIMITED,
+        ALL,
+        FORK,
+        FINALIZED
+    }
+
+    ReportingState private reportingState = ReportingState.AUTOMATED;
```

Comment:
The code does try to accommodate that bit by in some cases actually checking for `> LIMITED` rather than `==`. Additionally we usually use methods that ask if we're in a particular phase rather than just providing a getter to the state enum value. That said I understand this state value is confusing since it may be incorrect. I could rename the first value to reflect that it is somewhat ambiguous?